pyhap/characteristic.py

```diff
@@ -98,14 +98,39 @@ def __init__(self, display_name, type_id, properties):
         self.type_id = type_id
         self.properties = properties
         self.broker = None
+        self._value = None
+        self.getter_callback = None
         self.setter_callback = None
-        self.value = self._get_default_value()
 
     def __repr__(self):
         """Return the representation of the characteristic."""
         return '<characteristic display_name={} value={} properties={}>' \
             .format(self.display_name, self.value, self.properties)
 
+    @property
+    def value(self):
+        """
+        The value for the `Characteristic`.
+
+        `Characteristic.getter_callback`
+        You may define a `getter_callback` on the `Characteristic`
+        to override the returned value. `getter_callback` must
+        return a value. This will also set the `Characteristic`
+        value to the returned value from `getter_callback`.
+
+        To set the value for the `Characteristic` on a state
+        change, use `set_value` to handle notifying clients.
+
+        .. seealso:: Characteristic.set_value()
+
+        :return: value
+        """
+        if not getattr(self, '_value', None):
+            self._value = self._get_default_value()
+        if self.getter_callback:
+            self._value = self.to_valid_value(self.getter_callback())
```

Comment:
Maybe change that to `self._value = self.to_valid_value(value=self.getter_callback())`.
By adding `value=` it might be a bit clearer what `self.getter_callback()` will return.